presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java

```diff
@@ -160,8 +160,35 @@ public void testPlanningFailure()
         assertFailedQuery("SELECT * FROM mock.default.tests_table", "Throw from apply projection");
     }
 
+    @Test
+    public void testAbortedWhileWaitingForResources()
+            throws Exception
+    {
+        Session mySession = Session.builder(session)
+                .setSystemProperty("required_workers_count", "17")
+                .setSystemProperty("required_workers_max_wait_time", "10ms")
+                .build();
```

Comment:
nice

Reply:
Thank you :) This was the second attempt. I first tried to do testing without `QueryRunner`, setting up everything manually. And I failed miserably :)